app/src/main/java/org/jellyfin/mobile/data/entity/DownloadEntity.kt

```diff
@@ -28,6 +30,8 @@ data class DownloadEntity(
     @ColumnInfo(name = DOWNLOAD_LENGTH)
     val downloadLength: Long,
 ) {
+    fun asMediaSource() = decodeFromString<RemoteJellyfinMediaSource>(mediaSource)
```

Comment:
Why is this a RemoteJellyfinMediaSource? Wouldn't it make sense to return a LocalJellyfinMediaSource?

Reply:
That's absolutely right! 58128731445270ce496a2ef090b1f24685468130 is only a WIP and I haven't had time to go through all uses just yet. I'll get these changed asap when I have a moment to spare.

Haha don't worry, you aren't missing anything 😆

Reply:
I'm just really not that experienced with Android development, so yeeaah

Reply:
I've gone through what's left of the WIP and I believe all the `JellyfinMediaSource` type parameters are all _valid_ now. Please let me know if you have any more you'd like changing.

p.s., You and me both! This is my first proper android project 😆